spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/write/BigQueryWriteHelper.java

```diff
@@ -105,7 +124,23 @@ public void writeDataFrameToBigQuery() {
       String format = config.getIntermediateFormat().getDataSource();
       data.write().format(format).save(gcsPath.toString());
 
-      loadDataToBigQuery();
+      if (writeDisposition == JobInfo.WriteDisposition.WRITE_TRUNCATE
+          && config.getPartitionOverwriteModeValue() == PartitionOverwriteMode.DYNAMIC
+          && bigQueryClient.tableExists(config.getTableId())) {
+        temporaryTableId = createTempTable(config.getTableId(), tableSchema);
```

Comment:
`temporaryTableId = Optional.of(createTempTable(config.getTableId(), tableSchema));`